### DISCUSSION TITLE
Sonoff Mini R4 with ESP32, flash tasmota32.bin OK, but Sonoff won't start

### DISCUSSION THREAD

**Discussion by brunosso** [arendst/Tasmota#20692] [Q&A] [answered]
Hi!
I try to flash Tastmota 13.3.0 on a Sonoff Mini R4 device.
I open it, solder the 4 pin (3V3, TX, RX and GND) and flash via Tasmota-Esp-Flasher 2.2.3.

The log is correct (i delete the same row with Writing bytes info for shortness)
```
Using 'COM5' as serial port.
Connecting...
Detecting chip type... ESP32
Connecting...

Chip Info:
 - Chip Family: ESP32
 - Chip Model: ESP32-D0WDQ5-V3 (revision v3.0)
 - Number of Cores: 2
 - Max CPU Frequency: 240MHz
 - Has Bluetooth: YES
 - Has Embedded Flash: NO
 - Has Factory-Calibrated ADC: YES
 - MAC Address: B0:A7:32:D8:5A:EC
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 1500000
Changed.
 - Flash Size: 4MB
Creating esp32 image...
Successfully created esp32 image.
Erasing flash (this may take a while)...
Chip erase completed successfully in 1.9s
Flash will be erased from 0x00001000 to 0x00005fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x0000e000 to 0x0000ffff...
Flash will be erased from 0x00010000 to 0x000d0fff...
Flash will be erased from 0x000e0000 to 0x002b8fff...
Compressed 17904 bytes to 12445...
Writing at 0x00001000... (100 %)
Wrote 17904 bytes (12445 compressed) at 0x00001000 in 0.2 seconds (effective 589.4 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 134...
Writing at 0x00008000... (100 %)
Wrote 3072 bytes (134 compressed) at 0x00008000 in 0.0 seconds (effective 1085.9 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 47...
Writing at 0x0000e000... (100 %)
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.0 seconds (effective 1492.2 kbit/s)...
Hash of data verified.
Compressed 786656 bytes to 560740...
Writing at 0x00010000... (2 %)
Writing at 0x00016930... (5 %)
Writing at 0x0001f2a4... (8 %)
...
Writing at 0x000c8db3... (97 %)
Writing at 0x000ceac6... (100 %)
Wrote 786656 bytes (560740 compressed) at 0x00010000 in 7.7 seconds (effective 820.1 kbit/s)...
Hash of data verified.
Compressed 1933584 bytes to 1214978...
Writing at 0x000e0000... (1 %)
Writing at 0x000e7af0... (2 %)
Writing at 0x000f00bf... (4 %)
...
Writing at 0x002ac084... (97 %)
Writing at 0x002b139c... (98 %)
Writing at 0x002b7222... (100 %)
Wrote 1933584 bytes (1214978 compressed) at 0x000e0000 in 17.4 seconds (effective 888.5 kbit/s)...
Hash of data verified.

Leaving...
Hard Resetting...
Hard resetting via RTS pin...
Done! Flashing is complete!

Showing logs:
Serial port closed!
```

bu now when i desolder the 4 flashing pin, and connect 230V to Lin and N the led onboard won't turn ON.
Even the relays override with the button don't work.

I think the firmware is correct, because the chip onboard is a ESP32, i can't use the tasmota.bin.

The PCB says it's the MINIR4 V1.0 with data code 2022.10.10

Someone can help me?

**Comment by GijsTimmers** [comment]
I have the same device, I got it working using the [Tasmota Web Installer](https://tasmota.github.io/install/). IIRC I just used _Tasmota (English)_ and it figured out how to flash it. I am running Tasmota 13.3.0(tasmota32) on it.

**Comment by Noschvie** [comment]
For initial serial flashing you have to use tasmota32.factory.bin

**Comment by brunosso** [comment, marked as answer]
I find the problem!

3,3V line from the flasher is poor of current.
After the flash with the same 3v3 from the flasher i restart the device with an external 3v3 source from a AMS1117-3.3...

Now i can find the AP...

Thanks everyone!

**Reply by DayBit** [reply]
I have the same problem, but I don't understand why doing that solved the problem. Isn't it equivalent to just turning MINIR4 off and on (reconnecting the usb) without pressing the button? Removing the pins and connecting 230V keeps the MINIR4 in programming mode?

**Reply by sfromis** [reply]
Many USB/serial adapters are not able to deliver enough current for booting ESP32 to Wifi, and then it can help connecting a better 3.3V power source, or indeed disconnect the wires from flashing before powering via mains. It is **not** about keeping the device in programming mode, which is not a goal.

You may **instead** have an unsuitable USB/serial adapter not even able to supply the current for flashing (that's not rare, and the reason why we recommend certain ones). In this case it can help connecting a better 3.3V power source on the side, or get a better adapter.

**Comment by llopez** [comment]
I am having the same issue. Just bought a brand new sonoff mini r4 extreme from aliexpress.

I am using esptool.py for flashing

it detects ESP32-D0WD-V3

the flashing process completes successfully but when I connect the device to 220 line it does not start. No tasmota network appear

**Reply by sfromis** [reply]
I'd recommend using the [Tasmota Web Installer](https://tasmota.github.io/install/), less risk of errors like flashing the wrong file (you did not mention any details what you did).

**Reply by llopez** [reply]
thank you for been so fast in the response. you are right

I am flashing this versions -> https://ota.tasmota.com/tasmota32/release/tasmota32.bin

with this commands:

```
esptool.py --chip esp32 --port /dev/ttyUSB0 erase_flash
esptool.py --chip esp32 --port /dev/ttyUSB0 write_flash -z 0x0 tasmota32.bin
```
output:
```esptool.py v4.8.1
Serial port /dev/ttyUSB0
Connecting...
Chip is ESP32-D0WD-V3 (revision v3.1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: f8:b3:b7:3d:da:2c
Stub is already running. No upload is necessary.
Configuring flash size...
Flash will be erased from 0x00000000 to 0x001fcfff...
Compressed 2083568 bytes to 1295121...
Wrote 2083568 bytes (1295121 compressed) at 0x00000000 in 115.8 seconds (effective 144.0 kbit/s)...
Hash of data verified.
```

the flashing work perfectly but the devices does not work even connected to 220v

**Reply by llopez** [reply]
I did it. It worked with the tasmota installer web and also suppliying 3.3 v with a better power supply.

I could find the Ap and configured the wifi network.

now, I need to realize how to map the template

**Reply by llopez** [reply]
I also did it :)

I used this template -> https://templates.blakadder.com/sonoff_MINIR4.html

everything works as expected..

thanks for the help

**Reply by DayBit** [reply]
Hi,

What was wrong with your commands using esptool.py? wrong firmware? bad 3.3V? The tasmota installer web does something different?

Thanks in advance

**Reply by sfromis** [reply]
As already explained previously in this thread, using `tasmota32.bin` is the wrong file. You can find the files "for initial flashing using esptool" on https://ota.tasmota.com/tasmota32/release/

Or just use the Tasmota Web Installer to do it right.

**Reply by DayBit** [reply]
I have tried almost all (including the Tasmota Web Installer) and no AP when powered on mains. Two options: bad 3.3v or my device is broken.

**Comment by llopez** [comment]
Clarification on Post-Flashing Behavior (Sonoff Mini R4)

After flashing the firmware, it may not be immediately apparent that the device is operational. This can lead to confusion, especially during initial setup and testing.

Key considerations:

Power requirements: The device must be supplied with a stable 3.3V source capable of delivering sufficient current. If power is inadequate, the Access Point (AP) will not initialize, making the device appear unresponsive.

Lack of visual feedback: By default, the onboard LED remains off because GPIOs are not pre-configured. Until a valid template is assigned via the Tasmota web UI, the LED and other peripherals will remain inactive.

Recommendation:

Use the Tasmota Web Installer with the Tasmota (English) firmware build.

Ensure the power supply delivers consistent 3.3V with adequate current (≥500mA recommended).

Once the AP is visible, connect and assign the appropriate template for the Sonoff Mini R4:
👉 [Blakadder Template](https://templates.blakadder.com/sonoff_MINIR4.html)

This will correctly map the GPIOs and enable LED status indication, confirming proper operation.

**Reply by sfromis** [reply]
After flashing firmware, it is the "safer strategy" to disconnect the wiring used for flashing, and instead power via the intended power source, like mains.

**Reply by DayBit** [reply]
I have the same problem, flash is OK but no visible AP (on 220v). I don't understand. If the firmware was flashed correctly (I read it after writing and the checksum is the same), doesn't that mean the 3.3V power supply is enough? I have a lab power supply I could use to supply the 3.3V but... is it safe to use it at the same time than the PC's USB connection?

**Reply by sfromis** [reply]
Matching checksum does not mean that the flashing was as it should, as you might be flashing the wrong binary, or not having correct parameters. Much depends on details of how you attempted to flash.

**Reply by BBracing** [reply]
**Flashing tool:** Tasmota Web Installer
**Firmware used:** tasmota32.factory.bin (latest release from https://ota.tasmota.com/tasmota32/release/)

---

After flashing, the process completes successfully with no errors.
However, when powering the device (either with 3.3 V from the serial adapter or 230 V AC),
there is **no serial log output** at 115200 baud and **no Wi-Fi AP (tasmota-xxxx)** appears.

I have already tried:
- Flashing multiple times using the Web Installer
- Performing `erase_flash` before flashing
- Holding and releasing GPIO0 in different sequences
- Waiting more than 60 seconds after power-up
- Measuring voltage on 3V3 pin (steady 3.27 V)
- Disconnecting the serial adapter before powering from mains

Despite that, the Mini R4 shows no signs of life — no LED activity, no serial boot log, and no visible AP.

Has anyone experienced the same behavior after flashing `tasmota32.factory.bin` on a Mini R4 Extreme?
Could this device require a specific binary variant (e.g. bluetooth / lite) or a different flash offset?

Any help or working example would be appreciated.

**Reply by darkdan2000** [reply]
I had the exact same problem. Flashing went smoothly, but no AP was created for me either—neither when connected to the FTDI adapter nor when powered via AC. It almost drove me crazy.

I reflashed it about 100 times, even with different firmware versions. I also tried debugging via the serial console, but I got no output at all from the Tasmota web flasher console/log.

Using python -m serial.tools.miniterm COM7 115200, I was finally able to see that the MiniR4 wasn’t booting into the firmware at all—it was stuck in flash mode. GPIO0 / the boot pin was being held LOW permanently.

Long story short: in my case, the problem was that when soldering the cable for Ground, I accidentally bridged it with the adjacent pin (Key). So the Ground pin and the Key pin (which sits right next to it) were connected.

Because the pins are so close together and there was only a tiny amount of solder, it was very hard to spot visually. I only found the issue using a multimeter (continuity test).

So if anyone else can’t find an AP after flashing, you should definitely check your soldering for this issue.
<img width="334" height="460" alt="image" src="https://github.com/user-attachments/assets/aa049375-6167-464c-ad5b-31acb850f6b9" />
